subsys/nfc/lib/platform.c

```diff
@@ -190,3 +322,57 @@ void nfc_platform_event_handler(nrfx_nfct_evt_t const *event)
 		break;
 	}
 }
+
+
+#ifdef CONFIG_NFC_THREAD_CALLBACK
+void nfc_platform_cb_request(void *ctx,
+			     size_t ctx_len,
+			     const uint8_t *data,
+			     size_t data_len,
+			     bool copy_data)
+{
+	uint32_t item_buf_tx[MAX_ITEM_SIZE32];
+	size_t item_len;
+	uint8_t item_len32;
+	int err;
+	uint8_t *buf = (uint8_t *)item_buf_tx;
+
+	item_len = ctx_len;
+
+	memcpy(buf, ctx, ctx_len);
+
+	if (copy_data) {
+		memcpy(&buf[ctx_len], data, data_len);
+		item_len += data_len;
+	} else {
+		*((uint32_t *)(&buf[ctx_len])) = POINTER_TO_UINT(data);
+		item_len += sizeof(uint32_t);
+	}
+
+	item_len32 = ROUND_UP(item_len, sizeof(uint32_t));
+	item_len32 /= sizeof(uint32_t);
+
+	err = ring_buf_item_put(&nfc_cb_ring, (uint16_t)copy_data, (uint8_t)ctx_len, item_buf_tx,
```

Comment:
Seems like copy_data is a boolean type. Would not be better to invert the argument order here. I mean you can use value parameter for copy_data and type parameter for ctx length. Then the context wouldn't be limited to 255 bytes